File: dp-core/src/main/java/com/dataplatform/system/entity/OperationLog.java
package com.dataplatform.system.entity;

import java.util.Date;

/**
 * 操作日志实体
 */
public class OperationLog {
    private Long id;
    private Long userId;
    private String username;
    private String operationType;
    private String moduleName;
    private String operationDesc;
    private String requestMethod;
    private String requestUrl;
    private String requestParams;
    private String responseResult;
    private String ipAddress;
    private Long durationMs;
    private String status;
    private String errorMessage;
    private String beforeData;
    private String afterData;
    private Date createTime;

    // Getters and Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Long getUserId() { return userId; }
    public void setUserId(Long userId) { this.userId = userId; }
    public String getUsername() { return username; }
    public void setUsername(String username) { this.username = username; }
    public String getOperationType() { return operationType; }
    public void setOperationType(String operationType) { this.operationType = operationType; }
    public String getModuleName() { return moduleName; }
    public void setModuleName(String moduleName) { this.moduleName = moduleName; }
    public String getOperationDesc() { return operationDesc; }
    public void setOperationDesc(String operationDesc) { this.operationDesc = operationDesc; }
    public String getRequestMethod() { return requestMethod; }
    public void setRequestMethod(String requestMethod) { this.requestMethod = requestMethod; }
    public String getRequestUrl() { return requestUrl; }
    public void setRequestUrl(String requestUrl) { this.requestUrl = requestUrl; }
    public String getRequestParams() { return requestParams; }
    public void setRequestParams(String requestParams) { this.requestParams = requestParams; }
    public String getResponseResult() { return responseResult; }
    public void setResponseResult(String responseResult) { this.responseResult = responseResult; }
    public String getIpAddress() { return ipAddress; }
    public void setIpAddress(String ipAddress) { this.ipAddress = ipAddress; }
    public Long getDurationMs() { return durationMs; }
    public void setDurationMs(Long durationMs) { this.durationMs = durationMs; }
    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }
    public String getErrorMessage() { return errorMessage; }
    public void setErrorMessage(String errorMessage) { this.errorMessage = errorMessage; }
    public String getBeforeData() { return beforeData; }
    public void setBeforeData(String beforeData) { this.beforeData = beforeData; }
    public String getAfterData() { return afterData; }
    public void setAfterData(String afterData) { this.afterData = afterData; }
    public Date getCreateTime() { return createTime; }
    public void setCreateTime(Date createTime) { this.createTime = createTime; }
}

File: dp-ui/src/types/operationLog.ts
/**
 * 操作日志类型定义
 */

export interface OperationLog {
  id: number
  userId?: number
  username: string
  operationType: string
  moduleName: string
  operationDesc: string
  requestMethod: string
  requestUrl: string
  requestParams?: string
  responseResult?: string
  ipAddress: string
  durationMs: number
  status: 'success' | 'failed'
  errorMessage?: string
  createTime: string
}

export interface OperationLogQuery {
  userId?: number
  username?: string
  moduleName?: string
  operationType?: string
  status?: string
  startTime?: string
  endTime?: string
  keyword?: string
  page?: number
  pageSize?: number
}

export interface OperationStats {
  total: number
  successCount: number
  failedCount: number
  successRate: number
}

export interface UserStats {
  username: string
  count: number
  success_count: number
  failed_count: number
}

export interface ModuleStats {
  module_name: string
  count: number
}

export interface OperationTypeStats {
  operation_type: string
  count: number
}

export interface TrendData {
  time_key: string
  count: number
  success_count: number
  failed_count: number
}
